File: Training Platform/Java/src/main/java/frc/robot/RobotContainer.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

package frc.robot;

import frc.robot.gamepad.OI;
import frc.robot.subsystems.Training;
import frc.robot.commands.Drive;

public class RobotContainer {

  /**
   * Create the subsystems and gamepad objects
   */
  public static Training train;
  public static OI oi;

  public RobotContainer()
  {
      //Create new instances
      train = new Training();
      oi = new OI();
      
      //Set the default command for the training subsytem
      train.setDefaultCommand(new Drive());
  }
}

File: WSA 2022 Training/C++/src/main/cpp/RobotContainer.cpp
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

#include "RobotContainer.h"

RobotContainer::RobotContainer()
{
  training.SetDefaultCommand(ManualDrive(&training, &oi));

  //Add Commands to the auto chooser
  chooser.AddOption("Drive Lenght of Court", &m_lengthOfCourt);
  chooser.AddOption("Rotate 720", &m_cleaningPad);

  frc::Shuffleboard::GetTab("Autonomous Options").Add(chooser);
}

frc2::Command* RobotContainer::GetAutonomousCommand()
{
  return chooser.GetSelected();
}
